File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.py
class Solution(object):
    def processStr(self, s):
        """
        :type s: str
        :rtype: str
        """
        res=[]
        for ch in s:
            if ch.islower():
                res.append(ch)
            if ch=="*":
                if res:
                    res.pop()
            if ch=="#":
                res+=res
            if ch=='%':
                res.reverse()
        return "".join(res)

File: 3612-process-string-with-special-operations-i/3612-process-string-with-special-operations-i.java
class Solution {
    public String processStr(String s) {
        StringBuilder res=new StringBuilder();
        for(char ch:s.toCharArray()){
            if (Character.isLowerCase(ch)) res.append(ch);
            if (ch=='*')
                if (res.length()>0) res.deleteCharAt(res.length()-1);
            if (ch=='#') res.append(res.toString());
            if (ch=='%') res=res.reverse();
        }
        return res.toString();
    }
}